hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/utils/SignatureMapUtilsTest.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.test.utils;
+
+import static com.hedera.node.app.service.contract.impl.utils.SignatureMapUtils.preprocessEcdsaSignatures;
+import static com.hedera.node.app.service.contract.impl.utils.SignatureMapUtils.validChainId;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.hedera.hapi.node.base.SignatureMap;
+import com.hedera.hapi.node.base.SignaturePair;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for the SignatureMapUtils class.
+ */
+class SignatureMapUtilsTest {
+
+    @Test
+    void testPreprocessEcdsaSignatures() {
+        final var keyBytes = Bytes.wrap(new byte[65]);
+        final var sigMap = new SignatureMap.Builder()
+                .sigPair(new SignaturePair.Builder().ecdsaSecp256k1(keyBytes).build())
+                .build();
+        final var resultSigMap = preprocessEcdsaSignatures(sigMap, 0);
+        assertThat(resultSigMap.sigPair().get(0).hasEcdsaSecp256k1()).isTrue();
+        // Assert truncated to 64 bytes
+        assertThat(resultSigMap.sigPair().get(0).ecdsaSecp256k1().length()).isEqualTo(64);
+    }
+
+    @Test
+    void testPreprocessEcdsaSignaturesForEdSigs() {
+        final var keyBytes = Bytes.wrap(new byte[65]);
+        final var sigMap = new SignatureMap.Builder()
+                .sigPair(new SignaturePair.Builder().ed25519(keyBytes).build())
+                .build();
+        final var resultSigMap = preprocessEcdsaSignatures(sigMap, 0);
+        assertThat(resultSigMap.sigPair().get(0).hasEd25519()).isTrue();
+        // Assert not truncated
+        assertThat(resultSigMap.sigPair().get(0).ed25519().length()).isEqualTo(65);
+    }
+
+    @Test
+    void testPreprocessEcdsaSignaturesWrongChainId() {
+        final var chainId = 199;
+        final var ecSig = new byte[65];
+        ecSig[64] = (byte) (35 + (chainId * 3));
+        final var keyBytes = Bytes.wrap(ecSig);
+        final var sigMap = new SignatureMap.Builder()
+                .sigPair(new SignaturePair.Builder().ecdsaSecp256k1(keyBytes).build())
+                .build();
+        assertThrows(IllegalArgumentException.class, () -> preprocessEcdsaSignatures(sigMap, 0));
+    }
+
+    @Test
+    void testWrongChainId() {
+        final var chainId = 199;
+        final var ecSig = new byte[65];
+        ecSig[64] = (byte) (35 + (chainId * 3));
+        assertThat(validChainId(ecSig, chainId)).isFalse();
+    }
+
+    @Test
+    void testCorrectChainId() {
+        final var chainId = 199;
+        int v = 35 + (chainId * 2);
+        // distribute the v value across the last two bytes
+        final var ecSig = new byte[66];
+        ecSig[65] = (byte) (v & 0xFF);
+        v >>= 8;
```

Comment:
`>>=` is obviously fine but you might want to use `>>>=` instead for ultra-rarity points.

(Go Java!  C++ doesn't have this operator: you have to cast to unsigned, then shift, then cast back to signed.)